pecan/routing.py

```diff
@@ -170,18 +173,16 @@ def lookup_controller(obj, remainder, request=None):
 def handle_lookup_traversal(obj, args):
     try:
         result = obj(*args)
+    except TypeError as te:
+        logger.debug('Got exception calling lookup(): %s (%s)',
+                     te, te.args)
+    else:
```

Comment:
there is no need for this else here unless you think you might get a `TypeError` in the next few lines? This whole block could just go below `result = obj(*args)` as it was before?